server.py

```diff
@@ -12,21 +12,56 @@ def server():
     try:
         while True:
             conn, addr = server_socket.accept()
-            buffsize = 32
+            buffsize = 16
             msg = ""
             complete = False
             while not complete:
                 line = conn.recv(buffsize)
-                print line
+                msg = "{}{}".format(msg, line)
                 if len(line) < buffsize:
                     complete = True
-                msg = "{}{}".format(msg, line)
-            conn.sendall("You sent: {}".format(msg))
-            print "You received: {}".format(msg)
+            response = parse_request(msg)
+            try:
+                conn.sendall(response)
+            except TypeError:
+                conn.sendall(response_error('400', 'BAD REQUEST'))
             conn.close()
     except KeyboardInterrupt:
+        print "\nServer successfully shut down"
         server_socket.close()
 
 
+def response_ok(msg):
+    result = "HTTP/1.1 200 OK\r\n"
+    con_type = "Content-Type: text/plain\r\n"
+    body = "Content length: {}".format(len(msg))
+    return "{}{}{}".format(result, con_type, body)
+
+
+def response_error(error_code, error_msg):
+    error_type = "HTTP/1.1 {} ERROR\r\n".format(error_code)
+    con_type = "Content-Type: text/plain\r\n"
+    body = "ERROR {}, {}\r\n".format(error_code, error_msg)
+    return "{}{}{}".format(error_type, con_type, body)
```

Comment:
There is no blank line separating your body from the headers of the request.  You must have that blank line to be speaking proper HTTP.